verilog/analysis/verilog_linter_test.cc

```diff
@@ -379,5 +387,287 @@ TEST(VerilogLinterDocumentationTest, AllRulesMarkdown) {
   EXPECT_TRUE(absl::StrContains(stream.str(), "Enabled by default:"));
 }
 
+class ViolationFixerTest : public testing::Test {
+ public:
+  ViolationFixerTest() : config_() {
+    config_.UseRuleSet(RuleSet::kNone);
+    config_.TurnOn("forbid-consecutive-null-statements");
+    config_.TurnOn("no-trailing-spaces");
+    config_.TurnOn("posix-eof");
+  }
+
+ protected:
+  LinterConfiguration config_;
+
+  // Returns fixed content
+  std::pair<absl::Status, std::string> LintAnalyzeFixText(
+      absl::string_view content,
+      ViolationFixer::AnswerChooser answer_chooser) const {
+    const ScopedTestFile temp_file(testing::TempDir(), content);
+
+    // Run the analyzer to produce a syntax tree from source code.
+    const auto analyzer =
+        absl::make_unique<VerilogAnalyzer>(content, temp_file.filename());
+    const absl::Status status = ABSL_DIE_IF_NULL(analyzer)->Analyze();
+
+    const auto& text_structure = analyzer->Data();
+
+    const auto lint_result =
+        VerilogLintTextStructure(temp_file.filename(), config_, text_structure);
+
+    std::ostringstream diagnostics;
+    const verilog::ViolationFixer violation_fixer(content, &diagnostics,
+                                                  &config_, answer_chooser);
+    violation_fixer.HandleLintRuleStatuses(
+        lint_result.value(), text_structure.Contents(), temp_file.filename());
+
+    std::string fixed_content;
+    const auto ok = GetContents(temp_file.filename(), &fixed_content);
+    return {lint_result.status(), fixed_content};
+  }
+
+  void DoFixerTest(std::initializer_list<ViolationFixer::AnswerChoice> choices,
+                   absl::string_view fixed_code) const {
+    // Fixable violations, in order:
+    // :2:10: no-trailing-spaces
+    // :3:10: forbid-consecutive-null-statements
+    // :4:10: forbid-consecutive-null-statements
+    // :4:11: no-trailing-spaces
+    // :5:10: forbid-consecutive-null-statements
+    // :6:10: forbid-consecutive-null-statements
+    // :7:10: no-trailing-spaces
+    // :7:14: posix-eof
+    absl::string_view original_code =
+        "module m;    \n"
+        "  wire a;;\n"
+        "  wire b;;  \n"
+        "  wire c;;\n"
+        "  wire d;;\n"
+        "endmodule    ";
+
+    auto it = choices.begin();
+    const auto answer_chooser =
```

Comment:
If instead of `auto`, you provide the type here `AnswerChooser`, you probably don't need to declare the `-> return type` in the closure.

Always try to be specific.

Reply:
done